### DISCUSSION TITLE
@react-spring/* vs react-spring

### DISCUSSION THREAD

**Discussion by CharlieHess** [pmndrs/react-spring#1125] [Support] [answered]
I see about a 50/50 mix of examples using the namespaced version of `@react-spring` vs the (original?) `react-spring`, which seems to be a combination of the namespaced modules.

I'm unable to find documentation on the difference between these versions, and which one is recommended or deprecated. This is absolutely maddening when you are using something like `@react-spring/three` or `@react-spring/core` and try to adapt from an example like [this](https://github.com/drcmda/learnwithjason/blob/master/src/App.js#L4), only to find that the `useTransition` API has changed … dramatically. Can someone provide the story here? It seems like there are two parallel universes for this project on npm and those universes are diverging.

**Comment by lilpolymath** [comment]
AFAIK, @react-spring is for modules - packages that are yet or not meant to be published as standalone packages. "@react-spring" is used to group packages under the organization to prevent polluting the global namespace while react-spring refers to the animation library itself. 

If you notice none of these packages in the @react-spring namespace is published as standalone yet.

**Comment by CharlieHess** [comment]
Thanks for the clarification @lilpolymath, the situation I had got myself into was that I had some scenes using `@react/spring` modules at v9, and some using the real `react-spring` at v8. Once I upgraded `react-spring` to v9 and fixed the API usage, it started behaving the same as the v9 submodules, which I was then able to remove.

**Comment by aleclarson** [comment]
Currently, `react-spring@latest` points to v8, and `react-spring@next` points to v9.

The `react-spring` package is an umbrella package that contains all supported platforms, which leads to peer dependency warnings when installing. The `@react-spring/*` scope is for internal packages (eg: `@react-spring/core`) as well as platform-specific packages (`@react-spring/native`, `@react-spring/web`, etc). Contrary to what @lilpolymath claims, the latter are absolutely standalone.

**Reply by lilpolymath** [reply]
Alright, then thanks for clarifying.

**Comment by notgiorgi** [comment]
I'm building an ui library (let's call it `ui-kit`) using `react-native-web` which will be used by both web and RN apps. I'm confused which kind of imports should I use in the library. I couldn't find anything about this in docs.

This is what I'm thinking:

- Use `react-spring/native` imports in `ui-kit` code
- When importing `ui-kit` in web projects, make a webpack alias for `react-spring/native => react-spring/web`
- When importing `ui-kit` in RN projects, don't do anything extra

Basically similar to what RNW recommends: https://necolas.github.io/react-native-web/docs/setup/#bundler

**Comment by joshuaellis** [comment, marked as answer]
Use @react-spring/* for the selected target. `react-spring` is every target, and in `npm@7` you'll get errors because you probably won't have the target peer deps installed for all of them i.e `react-three-fiber` and `konva`.

**Comment by ssantana-redenos** [comment]
Im try to find another example for react native, using useSprings, but its hard to fine something. In fact I do not see any react spring on react native.
